### PR TITLE
fix(core,platform): fix a11y issues for slider

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
Closes SAP/fundamental-ngx#5582
Closes SAP/fundamental-ngx#5081

#### Please provide a brief summary of this pull request.
This PR fixes following issues:
1. Announcement of minimum and maximum value of slider -- fixed using aria-label
2.  slider values is read twice -- fixed.
3.  incorrect 'has popup' announced - fixed 
4. custom label in slider is not read -- fixed.
5. single mode/range mode is not announced - fixed.
6. title for slider handle -fixed
7. reading custom label boundaries - fixed

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [NA] update `README.md`
- [NA] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Review by Betrozov — APPROVED** [review]

File: apps/docs/src/app/platform/component-docs/platform-slider/examples/form-field/slider-form-field-example.component.html [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <fdp-form-group class="slider-example" [formGroup]="customForm">
     <fdp-form-field id="value1" label="Single mode">
-        <fdp-slider name="value5" formControlName="value1"> </fdp-slider>
+        <fdp-slider name="value5" formControlName="value1" ariaLabel="Single mode"> </fdp-slider>
```
Comment (kavya-b) [inline apps/docs/src/app/platform/component-docs/platform-slider/examples/form-field/slider-form-field-example.component.html:3]: should `name` also be `value1`?

Comment (kavya-b) [reply apps/docs/src/app/platform/component-docs/platform-slider/examples/form-field/slider-form-field-example.component.html:3]: also PR title can be changed to `fix(platform): fix a11y issues for slider` in accordance with having only present tense in the message.

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Chrome screen reader is reading "min" as "minute" for me.  Also I think we should go with something more verbose to explain min/max to the user, what's currently being read would simply confuse me.  Maybe "Slider, current value is ____, min value is ____, max value is ____"

**Comment by DeepakSap14** [comment]
> Chrome screen reader is reading "min" as "minute" for me. Also I think we should go with something more verbose to explain min/max to the user, what's currently being read would simply confuse me. Maybe "Slider, current value is ____, min value is ____, max value is ____"

@mikerodonnell89 
verbose for reading min/max is fixed now.
for `Chrome screen reader is reading "min" as "minute"`,  I am not able to reduce reading speed, it may seem when it is trying to read "min is" very fast pace. also there is no way, it can be controlled. because it it coming from aria-valuemin.

**Review by InnaAtanasova — APPROVED** [review]

**Review by InnaAtanasova — DISMISSED** [review]
.

**Review by InnaAtanasova — APPROVED** [review]

**Comment by puru-hk** [comment]
Below are Comments from the ATL team 

Issue 1:

OB: The minimum and maximum value are not read.
EB: Ideally, the entire relevant information available should be read on navigating to the sliders. i.e. on navigating to the sliders the information regarding the maximum and minimum value available should be read.

issue 2:
OB: On accessing the slider, the values are read as '0' on changing the values.
image
EB: ideally, On changing the values the visible and changed values should be read on increasing and decreasing .

Issue 3:
OB: On accessing the slider, i can still able to reproduce the issue.
image
EB: Ideally, on changing the values within a slider using up/down arrow keys, the information regarding the value should be read only once and no additional information should be read.

Issue 4:
EB: On Changing the values are not read  but read as 'Zero' value.
image
EB: Ideally, on changing the value within the slider, the information regarding the focused value and the changed value should be read.

Issue 5:
OB: On changing the value read as '0' on increasing and decreasing the values.
image
EB: Ideally, the changed value must be read.

[Note: On increasing and Decreasing the value are read as '0'.
So please fix the issue

URL:https://deploy-preview-5683--fundamental-ngx.netlify.app
JAWS 2021

**Review by kavya-b — APPROVED** [review]

File: libs/core/src/lib/slider/slider.component.html [hunk 2]
```diff
@@ -50,13 +50,23 @@
         [class.fd-slider__handle--lg]="cozy"
         [attr.tabindex]="disabled ? -1 : 0"
         role="slider"
-        aria-label="slider"
-        [fdPopoverTrigger]="singleSliderPopover"
-        [attr.aria-valuemin]="min"
-        [attr.aria-valuemax]="max"
-        [attr.aria-valuenow]="getValuenow(_position, _sliderValueTargets.SINGLE_SLIDER)"
+        [attr.aria-label]="
+            ariaLabel || 'slider minimum value is ' + minValue + ', maximum value is ' + maxValue
```
Comment (platon-rov) [inline libs/core/src/lib/slider/slider.component.html:54]: There should be i18n for the default label

**Review by platon-rov — APPROVED** [review]

**Review by mikerodonnell89 — APPROVED** [review]

**Comment by puru-hk** [comment]
**Issue 5081:**

Issue 1: Fixed

Issue 2: 
image
OB: Throughout the control, on changing the values read thrice to the user.
EB: On changing the value using up/down arrow keys the changed value should be read once to the user.
Note: throughout the control

Issue 3: 
image
OB: On accessing the slider as visually there is no pop up read as has pop up.
EB: ideally, such additional information such as 'Has pop up' should be removed.

Issue 4: Fixed

Issue 5: Fixed

Issue 6: Fixed

Issue 7: 
image
OB: on adapting the screen to320 CSS, the text gets truncated.
EB: Ideally, the texts shouldn't get truncated when screen is resized to 320 pixels. It should be clearly visible.

**Comment by puru-hk** [comment]
**Issue 5582:**

Issue 1: Fixed

Issue 2: 
image
The value being read thrice through out the application.

OB: On changing the values using up/down arrow key the values are read thrice to the user.
EB: Ideally, on changing the values within a slider using up/down arrow keys, the information regarding the value should be read only once.

Issue 3: 
image

OB: On changing the values using up/down arrow key the values are read thrice to the user.
EB: Ideally, on changing the values within a slider using up/down arrow keys, the information regarding the value should be read only once.

Issue 4:

image
OB: On changing the values using up/down arrow key the values are read thrice to the user.
EB: Ideally, on changing the values within a slider using up/down arrow keys, the information regarding the value should be read only once. 
The Range value should be read to the user

Issue 5:
image
OB: On changing the values using up/down arrow key the values are read thrice to the user.
EB: Ideally, on changing the values within a slider using up/down arrow keys, the information regarding the value should be read only once.

URL:https://deploy-preview-5683--fundamental-ngx.netlify.app
JAWS 2021

**Comment by puru-hk** [comment]
Issue 5081:
Issue 7 I have tested in the real device. It is working properly. You can ignore

**Comment by DeepakSap14** [comment]
@puru-hk 
please re-verify this one more time. some improvements are done as we had discussed and issue were raised.
for issue #5081, issue 3('Has pop up' should be removed.), this is correct. As we have popup there at slider handle as tooltip. this information will be read. there is no way, we will be able to remove that.

**Comment by puru-hk** [comment]
@DeepakSap14 Demo url is wrong. Could you please update the url..

**Comment by mikerodonnell89** [comment]
@DeepakSap14 e2e tests failing, looks like a legit error related to slider

**Review by Betrozov — APPROVED** [review]

File: libs/core/src/lib/slider/slider.component.ts [hunk 3]
```diff
@@ -551,6 +602,27 @@ export class SliderComponent implements OnInit, OnChanges, OnDestroy, ControlVal
...
+
+    /** @hidden reset default prefix on leaving the slider */
+    onBlur(event: MouseEvent | KeyboardEvent): void {
+        // hide popover when focus moves out of slider handle
+        this._hidePopovers();
+
+        // reset prefix string for slider current value that need to be announced
+        if (this._isRange) {
+            this._rangeSliderHandle2CurrentValuePrefix = this.rangeSliderHandle2CurrentValuePrefix;
+            this._rangeSliderHandle2CurrentValuePrefix = this.rangeSliderHandle2CurrentValuePrefix;
```
Comment (aberikashvili) [inline libs/core/src/lib/slider/slider.component.ts:619]: Please remove duplicated line

File: libs/core/src/lib/slider/slider.component.ts [hunk 4]
```diff
@@ -613,18 +693,25 @@ export class SliderComponent implements OnInit, OnChanges, OnDestroy, ControlVal
...
+        rangeLowerValue = rangeValue[0];
+        rangeHigherValue = rangeValue[1];
+
         if (this.customValues.length > 0) {
             const min = this.customValues[rangeValue[0]] || this.customValues[0];
             const max = this.customValues[rangeValue[1]] || this.customValues[this.customValues.length - 1];
 
             rangeValue = [min, max];
+            rangeLowerValue = rangeValue[0].label;
+            rangeHigherValue = rangeValue[1].label;
```
Comment (aberikashvili) [inline libs/core/src/lib/slider/slider.component.ts:712]:
```suggestion
            const min = this.customValues[rangeValue[0]] || this.customValues[0];
            const max = this.customValues[rangeValue[1]] || this.customValues[this.customValues.length - 1];

            rangeLowerValue = min.label;
            rangeHigherValue = max.label;
```

File: libs/core/src/lib/slider/slider.component.ts (hunk 3 above)
Comment (DeepakSap14) [reply libs/core/src/lib/slider/slider.component.ts:617]: fixed

**Comment by puru-hk** [comment]
On increasing and decreasing the changed values are read twice using up and down arrow key.
Its known JAWS issue. Rest all issues looks fine

**Review by puru-hk — APPROVED** [review]

**Pull request merged**